src/sql/src/pure.rs

```diff
@@ -796,7 +796,18 @@ async fn compile_proto(
 /// whether the specified AWS configuration is valid.
 async fn validate_aws_credentials(config: &AwsConfig) -> Result<(), anyhow::Error> {
     let config = config.load().await?;
-    let sts_client = mz_aws_util::sts::client(&config)?;
-    let _ = sts_client.get_caller_identity().send().await?;
+    let _ok = mz_aws_util::sts::client(&config)
+        .map_err(|e| {
```

Comment:
looks like `client`'s error return type is `anyhow::Error` -- do you think it'd make sense to do `mz_aws_util::sts::client(&config).context("Unable to build STS client to validate AWS credentials")` instead of manually mapping the error?

Not a biggie but seems a little more best practices for using `anyhow`?

Reply:
Yeah anyhow's causes aren't printed unless you use the alternative format specifier (`{:#}` instead of just `{}`). We currently almost always use plain display, see my other comment [here](https://github.com/MaterializeInc/materialize/pull/10125#discussion_r788844927), so IMO the only way we can make context work correctly is to change all our error messages to not display their causes, and instead to always use the alternate form when formatting user-visible error messages. Which would be better, but is IMO out of scope for this PR.

[playground](https://gist.github.com/494ec70a36597e8da9e1943c114c6cde)

```rust
use anyhow::{anyhow, Context};

fn err() -> anyhow::Result<()> {
    Err(anyhow!("oh no!"))
}

fn main() {
    let result = err().context("this is helpful").unwrap_err();
    println!("basic: {}", result);
    println!("expanded: {:#}", result);
}
```

prints:

```text
basic: this is helpful
expanded: this is helpful: oh no!
```

Reply:
Ahh TIL -- thanks for the explanation and playground link!

Reply:
> IMO the only way we can make context work correctly is to change all our error messages to not display their causes, and instead to always use the alternate form when formatting user-visible error messages. Which would be better, but is IMO out of scope for this PR.

Yes. We should do this! We've been putting it off for way too long. I would much rather spend a few hours fixing this once and for all than continue to accumulate tech debt like this. I'm going to work on a PR for that.

Reply:
I checked, and we already correctly use the `{:#}` specifier when printing the cause for this error, so we can use `.context` here like @cjubb39 suggested.